stripe/src/main/java/com/stripe/android/view/PaymentFlowViewPager.kt

```diff
@@ -0,0 +1,28 @@
+package com.stripe.android.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.MotionEvent
+import androidx.viewpager.widget.ViewPager
+
+class PaymentFlowViewPager @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    private val isSwipingAllowed: Boolean = false
+) : ViewPager(context, attrs) {
+    override fun onInterceptTouchEvent(event: MotionEvent?): Boolean {
+        return if (isSwipingAllowed) {
+            super.onInterceptTouchEvent(event)
+        } else {
+            false
+        }
+    }
+
+    override fun onTouchEvent(event: MotionEvent?): Boolean {
+        return if (isSwipingAllowed) {
```

Comment:
`return isSwipingAllowed && super.onTouchEvent(event)` alternatively :)

Reply:
I’m going to end up getting rid of this in ViewPager2 so I’ll leave as is